### DISCUSSION TITLE
Prisma error parsing attribute "@relation"

### DISCUSSION THREAD

**Discussion by mohrazzak** [prisma/prisma#22498] [Q&A]
### Question

I'm currently grappling with a challenge in my Prisma data model and would greatly appreciate your insights and expertise in overcoming it.

Here's a snippet of my schema:

```prisma
model TextContent {
  // .. other fields
  StoreName        Store[]       @relation("store_name")
  StoreDescription Store[]       @relation("store_description")
}

model Store {
  // .. other fields
  nameId        Int?
  name          TextContent?  @relation("store_name", fields: [nameId], references: [id], onDelete: SetNull)

  descriptionId Int?
  description   TextContent?  @relation("store_description", fields: [descriptionId], references: [id], onDelete: SetNull)
}
```

My primary question revolves around configuring the Prisma schema to ensure that when a `Store` record is deleted, the associated `TextContent` records (pertaining to name, description, and slogan) are either automatically deleted or properly disassociated.

However, upon attempting to implement cascading deletion by changing the `onDelete` option in `TextContent`:

```prisma
StoreName        Store[] @relation("store_name", onDelete: Cascade)
```

I encountered the following error:

```
Error parsing attribute "@relation": The relation field `StoreName` on Model `TextContent` must not specify the `onDelete` or `onUpdate` argument in the @relation attribute. You must only specify it on the opposite field `name` on model `Store`, or in case of a many-to-many relation, in an explicit join table.
```

I chose the `"store_name"` approach due to multiple one-to-many relations with the same table.

Any advice or guidance on resolving this error and achieving the desired cascading deletion would be immensely appreciated.

Thank you in advance for your assistance!

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: Windows
- Database:  MySQL
- prisma  5.1.1

**Comment by janpio** [comment]
The `OnDelete` can only be set on one side of the relation. If you want to influence the behavior in the other direction, you either need to remodel or add another relation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
